src/handlers/awsScim.test.ts

```diff
@@ -0,0 +1,605 @@
+            { op: 'replace', path: 'userName', value: 'bjensen' },
+            { op: 'replace', path: 'displayName', value: 'Babs Jensen' },
+            { op: 'replace', path: 'active', value: true },
+            {
+              op: 'replace',
+              path: 'emails',
+              value: [
+                {
+                  value: 'bjensen@example.com',
+                  primary: true,
+                  type: 'work',
+                },
+                {
+                  value: 'babs123@example.com',
+                  primary: false,
+                  type: 'personal',
+                },
+              ],
+            },
+          ],
+        },
+      })
+    );
+  });
+
+  it('should have operations for create and delete of members from groups in patch request', async () => {
+    const payload = {
+      id: 'a8a90f06-89fc-5633-9205-0f37699f0eb6',
+      schemas: ['some-schema'],
+      Operations: [
+        {
+          op: 'replace',
+          value: {
+            members: [
+              { value: '8a58f666-46a9-522b-b40a-484b09db59ec' },
+              { value: 'edf32347-6d27-533f-a8ee-2898c657a184' },
+              { value: '1ee5e1d3-40dc-5ae1-aa51-aca7a832ddea' },
+            ],
+          },
+        },
+      ],
+    };
+
+    const event = createApiGatewayEvent(
+      'patch',
+      '/test/scim/v2/Groups/a8a90f06-89fc-5633-9205-0f37699f0eb6',
+      payload,
+      {}
+    );
+
+    await handler(event);
+
+    expect(axios).toHaveBeenLastCalledWith(
+      expect.stringContaining(event.requestContext.http.path),
+      expect.objectContaining({
+        method: 'patch',
+        data: expect.objectContaining({
+          schemas: ['urn:ietf:params:scim:api:messages:2.0:PatchOp'],
+          Operations: [
+            {
+              op: 'add',
+              path: 'members',
+              value: [
+                { value: 'edf32347-6d27-533f-a8ee-2898c657a184' },
+                { value: '1ee5e1d3-40dc-5ae1-aa51-aca7a832ddea' },
+              ],
+            },
+            {
+              op: 'remove',
+              path: 'members',
+              value: [{ value: '98feceb2-1ea1-5a8a-b818-4eb19c32166a' }],
+            },
+          ],
+        }),
+      })
+    );
+  });
+
+  it('should modify headers for upstream server', async () => {
+    const event = createApiGatewayEvent('get', 'test', undefined, {
+      'content-length': '100',
+      host: 'https://api-gateway-url',
+      authorization: 'Bearer aCoolToken',
+    });
+
+    await handler(event);
+
+    expect(axios).toHaveBeenLastCalledWith(
+      expect.stringContaining(event.requestContext.http.path),
+      expect.objectContaining({
+        headers: {
+          authorization: event.headers.authorization,
+          // @ts-ignore
+          host: new URL(process.env.PROXY_URL).hostname,
+        },
+      })
+    );
+  });
+
+  it('should respond with downstream statuses', async () => {
+    const statusesCodes = [
```

Comment:
Prefer that to define data outside the test case and then use a data driven test.

Reply:
Move a lot of the tests to data-driven tests. There are still a couple that has data inside the test case.